### DISCUSSION TITLE
Classification of subsequences or spans of tokens

### DISCUSSION THREAD

**Discussion by JuanFF** [sunglasses-ai/classy#89] [Q&A] [answered]
Hi,
Currently there are five tasks supported by Classy (sequence classification, sentence-pair classification, token classification, text extraction, and text generation). I need to generate a model to classify a group of tokens (or a subsequence) within a sequence, for example:

`hello there my name is juan what's your name `
where
`hello there` -> label X
`my name is juan` -> label Y
`what's your name` -> label Z

I'm wondering if token classification is the right approach here, as I would rather use it for NER (entity identification). These spans of tokens are not entities but rather sequences within a sequence. Would you use another task instead?

Thanks!

**Comment by edobobo** [comment, marked as answer]
Hi Juan,

Actually I would still use the token-level classification task with the BILOU formulation. 
So your sequence of tokens would be labeled like this:

- hello -> B-X
- there -> L-X
- my -> B-Y
- name -> I-Y
- is -> I-Y
- Juan -> L-Y
- what -> B-Z
- 's -> I-Z
- your -> I-Z
- name -> L-Z

hope this will help you! 
Out of curiosity, what are you using Classy for?

**Reply by JuanFF** [reply]
Thanks! I'm working on (transcribed) voice data, where I receive a long speech as input without inner boundaries. Unlike written text, there is no punctuation to split what the speaker says into different ideas. 
The appropriate task would be sequence labeling, but I'm forced to use token labeling because these sequences aren't self contained as in written texts.

**Reply by edobobo** [reply]
Very interesting, thanks for sharing!
